### ISSUE TITLE
Refactor supported regions, large anonymity country codes and other country codes to use std::set

### ISSUE THREAD

**Issue by tmancey** [brave/brave-browser#9205]
Refactor supported regions, large anonymity country codes and other country codes to use `std::set` for:

Change `const std::map<int, std::vector<std::string>> kSupportedRegionsSchemas` to `const std::map<int, std::set<std::string>> kSupportedRegionsSchemas`

Change `const std::map<std::string, bool> kLargeAnonymityCountryCodes` to `const std::set<std::string> kLargeAnonymityCountryCodes`

Change `const std::map<std::string, bool> kOtherCountryCodes` to `const std::set<std::string> kOtherCountryCodes`

**Comment by kjozwiak** [comment]
Verification **PASSED** on `macOS 10.15.5 x64` using the following build:

```
Brave | 1.10.90 Chromium: 83.0.4103.97 (Official Build) (64-bit)
-- | --
Revision | 326d148b9655369b86498d9ecca39f63dd2bdd2d-refs/branch-heads/4103@{#657}
OS | macOS Version 10.15.5 (Build 19F101)
```

* ensured that you can enable/view ads in the following supported regions
  * `Canada`, `USA`, `Poland`, `Russia`
* ensured that ads are disabled for unsupported regions like `Iran`

<img width="664" alt="Screen Shot 1399-03-23 at 15 35 46" src="https://user-images.githubusercontent.com/2602313/84542329-3c65df00-acc7-11ea-8203-7c902b0f04f4.png">

Ensured that `"buildChannel":"release"` and `"platform":"macos"` are being appended in `/v1/confirmation/{confirmation_id}/{credential}` requests as per the following:

```
INFO:redeem_token.cc(111)] CreateConfirmation
INFO:redeem_token.cc(113)] POST /v1/confirmation/{confirmation_id}/{credential}
INFO:redeem_token.cc(116)] URL Request:
INFO:redeem_token.cc(119)]   URL: https://ads-serve.brave.com/v1/confirmation/333c3ace-0f51-4a9f-aab9-7a431d987d67/eyJwYXlsb2FkIjoie1wiYmxpbmRlZFBheW1lbnRUb2tlblwiOlwibmx3d2l5eWpiMGFQclhZUTE1aGVCVDlTcFJGdUpVRERQWjcvWVFlbmNEST1cIixcImJ1aWxkQ2hhbm5lbFwiOlwicmVsZWFzZVwiLFwiY3JlYXRpdmVJbnN0YW5jZUlkXCI6XCI3YWMyMGFiZS05ZDY0LTQ1ZjMtYjA5NS1mODJlMTExZWE0N2NcIixcInBheWxvYWRcIjp7fSxcInBsYXRmb3JtXCI6XCJtYWNvc1wiLFwidHlwZVwiOlwidmlld1wifSIsInNpZ25hdHVyZSI6IkJzbzE2aGp6V01qRFlpenJpeHVVNm5mUGlQcUhmaHloNXRUT1RkcGVHTmlma2JuZVptMHpPRXFoeTRnZDRzM0RLVStEeU9pSVREQ2dsdldrdEtmS0FBPT0iLCJ0IjoiZ3l2V0xROVNscVgxR3M1STdHZVdsRzRBMlpjcFNQVk03dHFUakdEOTQrUWt1SVdLdmlSZFNmUlZGV2t3aFN3bjRDTUduc2NzL3pJVnpYV21TaXgyZWc9PSJ9
INFO:redeem_token.cc(131)]   Body: {"blindedPaymentToken":"nlwwiyyjb0aPrXYQ15heBT9SpRFuJUDDPZ7/YQencDI=","buildChannel":"release","creativeInstanceId":"7ac20abe-9d64-45f3-b095-f82e111ea47c","payload":{},"platform":"macos","type":"view"}
INFO:redeem_token.cc(134)]   Headers:
INFO:redeem_token.cc(136)]     accept: application/json
INFO:redeem_token.cc(140)]   Content_type: application/json
```

As per the STR via https://github.com/brave/brave-core/pull/5067#issue-394103417, `/v1/confirmation/{confirmation_id}/{credential}` requests should have the following being listed:

* `"buildChannel":"release"`
* `"platform":"macos"`
* `CountryCode`

However, as per https://github.com/brave/brave-browser/issues/9823, the `CountryCode` isn;t available in `1.10.x` and will be fixed in `1.11.x`.

**Comment by kjozwiak** [comment]
@jsecretan @tmancey assuming `CountryCode` missing from `1.10.x` and being included in `1.11.x` is fine?
